.github/workflows/test.yml

```diff
@@ -18,47 +18,37 @@ jobs:
           - name: latest
             version: latest
         ruby_version: ["3.2.2"]
-    name: Rails tests (RubyGems ${{ matrix.rubygems.name }})
+        tests:
+          - name: general
+            command: test
+          - name: system
+            command: test:system
+    name: Rails tests ${{ matrix.tests.name }} (RubyGems ${{ matrix.rubygems.name }})
     runs-on: ubuntu-22.04
     env:
       RUBYGEMS_VERSION: ${{ matrix.rubygems.version }}
-      # Fail hard when Toxiproxy is not running to ensure all tests (even Toxiproxy optional ones) are passing
       REQUIRE_TOXIPROXY: true
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
```

Comment:
probably makes sense to move the checkout step into the new action

Reply:
IMHO checkout is not part of "setup", but I can move it in there if you feel strongly about this.